pkg/namespacestore/reconciler.go

```diff
@@ -395,9 +400,65 @@ func (r *Reconciler) ReconcileDeletion() error {
 		}
 	}
 
+	if r.NamespaceStore.Spec.NSFS != nil {
+		pvcName := r.NamespaceStore.Spec.NSFS.PvcName
+		if !util.KubeCheck(r.DeploymentEndpoint) {
+			r.Logger.Infof("No Endpoint deployment yet. requeuing for 5 minutes")
+			return fmt.Errorf("No Endpoint Deployment. requeue again")
+		}
+		n := 0
+		if r.shouldCleanEndpointDeploymentPvc() {
```

Comment:
I would add here comments on removing volumes and mounts

Reply:
changed all logic - not relevant